lib/ansible/modules/cloud/vmware/vmware_host_sriov.py

```diff
@@ -0,0 +1,387 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: vmware_host_sriov
+short_description: Manage SR-IOV settings on host
+description:
+- This module can be used to configure, enable/disable SR-IOV functions on ESXi host.
+- Module doesn't  reboot the host after changes, but puts it in output "rebootRequired" state.
+- User can specify an ESXi hostname or Cluster name. In case of cluster name, all ESXi hosts are updated.
+version_added: "2.10"
+author:
+- Viktor Tsymbalyuk (@victron)
+notes:
+- Tested on vSphere 6.0
+requirements:
+- python >= 2.7
+- PyVmomi
+options:
+  esxi_hostname:
+    description:
+    - Name of the host system to work with.
+    - This parameter is required if C(cluster_name) is not specified.
+    - user can specify specific host from cluster.
+    type: str
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This parameter is required if C(esxi_hostname) is not specified.
+    type: str
+  vmnic:
+    description:
+    - interface name, like vmnic0
+    type: str
+    required: True
+  num_virt_func:
+    description:
+    - number of functions to activate on interface.
+    - 0 means SR-IOV disabled.
+    - number greater than 0 means SR-IOV enabled.
+    type: int
+    required: True
+  sriov_on:
+    description:
+    - optional parameter, related to num_virt_func.
+    - SR-IOV can be enabled only if num_virt_func > 0.
```

Comment:
SR-IOV can be enabled only if `C(num_virt_func) > 0.`

Reply:
ok, thanks